File: WhatIsForDinnerBackEnd/WhatIsForDinnerBackEnd/Models/Recipe.cs
﻿
using System.Diagnostics;

public class RecipeSearchResult
{
    public Result[] results { get; set; }
    public int offset { get; set; }
    public int number { get; set; }
    public int totalResults { get; set; }
}

public class Result
{
    public int id { get; set; }
    public string title { get; set; }
    public string image { get; set; }
    public string imageType { get; set; }
    public bool isFavorited { get; set; } = false;
    public Nutrition nutrition { get; set; }
}

//public class Nutrition
//{
//    public Nutrient[] nutrients { get; set; }
//}

//public class Nutrient
//{
//    public string name { get; set; }
//    public float amount { get; set; }
//    public string unit { get; set; }
//}
public class Recipe
{
    public bool vegetarian { get; set; }
    public bool vegan { get; set; }
    public bool glutenFree { get; set; }
    public bool dairyFree { get; set; }
    public bool veryHealthy { get; set; }
    public bool cheap { get; set; }
    public bool veryPopular { get; set; }
    public bool sustainable { get; set; }
    public bool lowFodmap { get; set; }
    public int weightWatcherSmartPoints { get; set; }
    public string gaps { get; set; }
    public int preparationMinutes { get; set; }
    public int cookingMinutes { get; set; }
    public int aggregateLikes { get; set; }
    public int healthScore { get; set; }
    public string creditsText { get; set; }
    public string license { get; set; }
    public string sourceName { get; set; }
    public float pricePerServing { get; set; }
    public Extendedingredient[] extendedIngredients { get; set; }
    public int id { get; set; }
    public string title { get; set; }
    public int readyInMinutes { get; set; }
    public int servings { get; set; }
    public string sourceUrl { get; set; }
    public string image { get; set; }
    public string imageType { get; set; }
    public string summary { get; set; }
    public object[] cuisines { get; set; }
    public string[] dishTypes { get; set; }
    public object[] diets { get; set; }
    public object[] occasions { get; set; }
    public Winepairing winePairing { get; set; }
    public string instructions { get; set; }
    public AnalizedInstructions[] analyzedInstructions { get; set; }
    public object originalId { get; set; }
    public string spoonacularSourceUrl { get; set; }
}

public class Winepairing
{
    public object[] pairedWines { get; set; }
    public string pairingText { get; set; }
    public object[] productMatches { get; set; }
}

public class Extendedingredient
{
    public int id { get; set; }
    public string aisle { get; set; }
    public string image { get; set; }
    public string consistency { get; set; }
    public string name { get; set; }
    public string nameClean { get; set; }
    public string original { get; set; }
    public string originalName { get; set; }
    public float amount { get; set; }
    public string unit { get; set; }
    public string[] meta { get; set; }
    public Measures measures { get; set; }
}

public class Measures
{
    public Us us { get; set; }
    public Metric metric { get; set; }
}

public class Us
{
    public float amount { get; set; }
    public string unitShort { get; set; }
    public string unitLong { get; set; }
}

public class Metric
{
    public float amount { get; set; }
    public string unitShort { get; set; }
    public string unitLong { get; set; }
}

public class AnalizedInstructions
{
    public string name { get; set; }
    public Step[] steps { get; set; }
}

public class Step
{
    public int number { get; set; }
    public string step { get; set; }
}

File: WhatIsForDinnerFrontEnd/src/app/recipe.ts


export enum Consistency {
    Liquid = "LIQUID",
    Solid = "SOLID",
}

export interface Measures {
    us:     Metric;
    metric: Metric;
}

export interface Metric {
    amount:    number;
    unitShort: string;
    unitLong:  string;
}

export interface WinePairing {
    pairedWines:    any[];
    pairingText:    string;
    productMatches: any[];
}

// Converts JSON strings to/from your types
export class Convert {
    public static toRecipe(json: string): Recipe {
        return JSON.parse(json);
    }

    public static recipeToJson(value: Recipe): string {
        return JSON.stringify(value);
    }
}

// To parse this data:
//
//   import { Convert, RecipeSearchResult } from "./file";
//
//   const recipeSearchResult = Convert.toRecipeSearchResult(json);

export interface RecipeSearchResult {
    results:      Result[];
    offset:       number;
    number:       number;
    totalResults: number;
}

export interface Result {
    id:        number;
    title:     string;
    image:     string;
    imageType: ImageType;
    nutrition: null;
    isFavorited:boolean;
}

export enum ImageType {
    Jpg = "jpg",
}


// To parse this data:
//
//   import { Convert } from "./file";
//
//   const empty = Convert.toEmpty(json);

export interface Recipe {
    vegetarian:               boolean;
    vegan:                    boolean;
    glutenFree:               boolean;
    dairyFree:                boolean;
    veryHealthy:              boolean;
    cheap:                    boolean;
    veryPopular:              boolean;
    sustainable:              boolean;
    lowFodmap:                boolean;
    weightWatcherSmartPoints: number;
    gaps:                     string;
    preparationMinutes:       number;
    cookingMinutes:           number;
    aggregateLikes:           number;
    healthScore:              number;
    creditsText:              string;
    sourceName:               string;
    pricePerServing:          number;
    extendedIngredients:      ExtendedIngredient[];
    id:                       number;
    title:                    string;
    readyInMinutes:           number;
    servings:                 number;
    sourceUrl:                string;
    image:                    string;
    imageType:                string;
    summary:                  string;
    cuisines:                 string[];
    dishTypes:                string[];
    diets:                    any[];
    occasions:                any[];
    winePairing:              WinePairing;
    instructions:             string;
    analyzedInstructions:     AnalyzedInstructions[];
    report:                   null;
    tips:                     Tips;
    openLicense:              number;
    suspiciousDataScore:      number;
    approved:                 number;
    unknownIngredients:       any[];
    userTags:                 any[];
    originalId:               null;
    spoonacularSourceUrl:     string;
    license?:                 string;
}

export interface AnalyzedInstructions {
    name:  string;
    steps: Step[];
}

export interface Step {
    number:      number;
    step:        string;
    ingredients: Ent[];
    equipment:   Ent[];
    length:      Length;
}

export interface Ent {
    id:            number;
    name:          string;
    localizedName: string;
    image:         string;
}

export interface Length {
    number: number;
    unit:   string;
}

export interface ExtendedIngredient {
    id:           number;
    aisle:        string;
    image:        string;
    consistency:  Consistency;
    name:         string;
    nameClean:    string;
    original:     string;
    originalName: string;
    amount:       number;
    unit:         string;
    meta:         string[];
    measures:     Measures;
    price:number;
    priceUnit:string;
}



export interface Measures {
    us:     Metric;
    metric: Metric;
}

export interface Metric {
    amount:    number;
    unitShort: string;
    unitLong:  string;
}

export interface Tips {
    health:  string[];
    price:   string[];
    cooking: string[];
    green:   string[];
}



